### PR TITLE
Celo mainnet branch

### LINKED ISSUE #45: Backend: Refactor api for celo mainnet

### LINKED ISSUE #47: Deploy safe and module on mainnet

### PR DESCRIPTION
**Pull request by venoivankovic**
init, mainnet deployment, tested w all milestones, worked
this is only the celo mainnet, not both testnet and mainnet

### CODE REVIEW

File: apps/Wallet/src/app/api/safe/add-user/route.ts [hunk 1]
```diff
@@ -2,13 +2,13 @@ import { UserManagementTransactionBody } from "@/types";
...
 import { Rbac__factory } from "../../../../types/typechain/types/config/abis";
-import { getSafe, getSigner } from "../util/utils";
+import { getCeloSigner, getSafe } from "../util/utils";
 
 export async function POST(req: Request): Promise<Response> {
   const body = (await req.json()) as UserManagementTransactionBody;
 
-  const safeSdk = await getSafe(process.env.OWNER_1_PRIVATE_KEY_GOERLI);
-  const rbacModuleAddress = process.env.RBAC_MODULE_ADDRESS!;
+  const safeSdk = await getSafe(process.env.OWNER_1_PRIVATE_KEY_CELO);
```
Comment (GabrielKidane96) [inline apps/Wallet/src/app/api/safe/add-user/route.ts]: pk should come from the body

File: apps/Wallet/src/app/api/safe/util/utils.ts [hunk 2]
```diff
@@ -31,11 +42,11 @@ export function getEtherscanSigner(pk: string) {
   return new ethers.Wallet(pk, provider);
 }
 
-function getEthAdapter(pk?: string): EthersAdapter {
-  const provider = new ethers.providers.JsonRpcProvider(process.env.RPC_URL!);
+async function getEthAdapter(pk?: string): Promise<EthersAdapter> {
```
Comment (GabrielKidane96) [inline apps/Wallet/src/app/api/safe/util/utils.ts:45]: not an eth adapter anymore then

File: apps/Wallet/src/features/web3auth/web3auth-provider.tsx [hunk 3]
```diff
@@ -11,17 +11,17 @@ export function Web3AuthProvider({ children }: { children: React.ReactNode }) {
   const web3Auth = useWalletStore((state) => state.web3Auth);
   const setWeb3Auth = useWalletStore((state) => state.setWeb3Auth);
   const clientId = process.env.NEXT_PUBLIC_clientId!;
-
+// changes need to be made here
```
Comment (GabrielKidane96) [inline apps/Wallet/src/features/web3auth/web3auth-provider.tsx]: When?

File: deploy-safe/util/constants.ts [hunk 4]
```diff
@@ -1,22 +1,19 @@
 // https://chainlist.org/?search=goerli&testnets=true
 export const RPC_URL_GOERLI = "https://eth-goerli.public.blastapi.io";
-export const RPC_URL_ALFAJORES = "https://alfajores-forno.celo-testnet.org";
+export const RPC_URL_CELO = "https://forno.celo.org"; // done
```
Comment (GabrielKidane96) [inline deploy-safe/util/constants.ts]: done?

File: deploy-safe/util/constants.ts [hunk 5]
```diff
@@ -1,22 +1,19 @@
...
 
 export const txServiceUrl_GOERLI = "https://safe-transaction-goerli.safe.global/";
-export const txServiceUrl_ALFAJORES = "https://client-gateway.celo-safe-prod.celo-networks-dev.org";
+export const txServiceUrl_CELO = "https://safe-transaction-celo.safe.global/"; // done
```
Comment (GabrielKidane96) [inline deploy-safe/util/constants.ts]: done?

File: apps/Wallet/src/app/api/safe/add-user/route.ts (hunk 1 above)
Comment (venoivankovic) [reply apps/Wallet/src/app/api/safe/add-user/route.ts]: fixed

File: apps/Wallet/src/features/web3auth/web3auth-provider.tsx (hunk 3 above)
Comment (venoivankovic) [reply apps/Wallet/src/features/web3auth/web3auth-provider.tsx]: fixed

File: deploy-safe/util/constants.ts (hunk 4 above)
Comment (venoivankovic) [reply deploy-safe/util/constants.ts]: fixed

File: deploy-safe/util/constants.ts (hunk 5 above)
Comment (venoivankovic) [reply deploy-safe/util/constants.ts]: fixed

**Pull request closed without merging**